dao/src/main/java/com/epam/esm/dao/impl/CertificateDaoImpl.java

```diff
@@ -0,0 +1,102 @@
+package com.epam.esm.dao.impl;
+
+import com.epam.esm.dao.CertificateDao;
+import com.epam.esm.dao.specification.Specification;
+import com.epam.esm.entity.Certificate;
+import org.springframework.beans.factory.annotation.Autowired;
+import org.springframework.jdbc.core.BeanPropertyRowMapper;
+import org.springframework.jdbc.core.JdbcTemplate;
+import org.springframework.stereotype.Component;
+
+import java.util.List;
+import java.util.Optional;
+
+@Component
+public class CertificateDaoImpl implements CertificateDao {
+    private final JdbcTemplate jdbcTemplate;
+
+    private final String FIND_BY_ID = """
```

Comment:
I believe these constants should be static.

Reply:
Fixed